src/main/java/net/minecraftforge/event/entity/EntityEvent.java

```diff
@@ -135,4 +135,20 @@ public EnteringChunk(Entity entity, int newChunkX, int newChunkZ, int oldChunkX,
         public int getOldChunkZ() { return oldChunkZ; }
         public void setOldChunkZ(int oldChunkZ) { this.oldChunkZ = oldChunkZ; }
     }
+    
+    /**
+     * This event is fired whenever an entity is about to tick in {@link World#updateEntityWithOptionalForce(net.minecraft.entity.Entity, boolean)} <br>
```

Comment:
Well theirs already LivingUpdateEvent and as far as I can tell it doesn't cause any additional lag so I think this would be alright.